pygmt/src/plot.py

```diff
@@ -206,15 +210,18 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
     kind = data_kind(data, x, y)
 
     extra_arrays = []
-    if "S" in kwargs and kwargs["S"][0] in "vV" and direction is not None:
-        extra_arrays.extend(direction)
+    if "S" in kwargs and isinstance(kwargs["S"], str):
```

Comment:
I added `isinstance(kwargs["S"], str)` in case that the `style` parameter is **True**, **False**, **None**, **int**, or **float**.